File: app/src/main/java/com/kingassistant/ScreenCaptureService.java
package com.kingassistant;

import android.app.Service;
import android.content.Intent;
import android.hardware.display.VirtualDisplay;
import android.media.projection.MediaProjection;
import android.media.projection.MediaProjectionManager;
import android.os.IBinder;
import android.view.WindowManager;

/**
 * এই সার্ভিসটি ফোনের স্ক্রিনকে রিয়েল-টাইমে ক্যাপচার করার জন্য তৈরি।
 * এটি সরাসরি MediaProjection API ব্যবহার করে ফোনের ডিসপ্লে স্ট্রিম করবে।
 * পরবর্তী ধাপে এই স্ট্রিম থেকে ডেটা নিয়ে এআই অ্যানালাইসিস করবে।
 */
public class ScreenCaptureService extends Service {

    private MediaProjection mediaProjection;
    private VirtualDisplay virtualDisplay;
    private MediaProjectionManager mediaProjectionManager;

    @Override
    public void onCreate() {
        super.onCreate();
        // স্ক্রিন ক্যাপচারের জন্য ম্যানেজার ইনিশিয়ালাইজ করা হচ্ছে
        mediaProjectionManager = (MediaProjectionManager) getSystemService(MEDIA_PROJECTION_SERVICE);
    }

    public void startCapture(int resultCode, Intent data) {
        mediaProjection = mediaProjectionManager.getMediaProjection(resultCode, data);
        
        // ভার্চুয়াল ডিসপ্লে তৈরি যা স্ক্রিন রিড করবে
        virtualDisplay = mediaProjection.createVirtualDisplay(
                "KingAssistantDisplay",
                1080, 1920, 400, // ফোনের রেজোলিউশন ও ডিপিআই (এটি আপনার ফোনের অনুযায়ী অ্যাডজাস্ট করা যাবে)
                0, null, null, null
        );
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (virtualDisplay != null) virtualDisplay.release();
        if (mediaProjection != null) mediaProjection.stop();
    }
}

File: ​app/src/main/java/com/pikachu/ScreenCaptureService.kt
package com.pikachu

import android.app.Service
import android.content.Intent
import android.hardware.display.DisplayManager
import android.hardware.display.VirtualDisplay
import android.media.projection.MediaProjection
import android.media.projection.MediaProjectionManager
import android.os.IBinder
import android.util.DisplayMetrics

/**
 * পিকাজু স্ক্রিন ক্যাপচার সার্ভিস
 * এটি MediaProjection API ব্যবহার করে রিয়েল-টাইম স্ক্রিন ক্যাপচার করবে।
 */
class ScreenCaptureService : Service() {

    private var mediaProjection: MediaProjection? = null
    private var virtualDisplay: VirtualDisplay? = null
    private var mediaProjectionManager: MediaProjectionManager? = null

    override fun onCreate() {
        super.onCreate()
        // সিস্টেম সার্ভিস থেকে MediaProjectionManager ইনিশিয়ালাইজ করা
        mediaProjectionManager = getSystemService(MEDIA_PROJECTION_SERVICE) as MediaProjectionManager
    }

    /**
     * ক্যাপচার শুরু করার জন্য মূল মেথড
     * @param resultCode অডিও-ভিজ্যুয়াল পারমিশন রেজাল্ট কোড
     * @param data পারমিশন ইনটেন্ট ডেটা
     */
    fun startCapture(resultCode: Int, data: Intent) {
        mediaProjection = mediaProjectionManager?.getMediaProjection(resultCode, data)
        
        // ভার্চুয়াল ডিসপ্লে তৈরি করা (পিকাজুর প্রসেসিংয়ের জন্য)
        val metrics = resources.displayMetrics
        virtualDisplay = mediaProjection?.createVirtualDisplay(
            "PikachuDisplay",
            metrics.widthPixels,
            metrics.heightPixels,
            metrics.densityDpi,
            DisplayManager.VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR,
            null, null, null
        )
    }

    override fun onBind(intent: Intent?): IBinder? {
        return null
    }

    override fun onDestroy() {
        super.onDestroy()
        // সার্ভিস বন্ধ হওয়ার সময় রিসোর্স রিলিজ করা
        virtualDisplay?.release()
        mediaProjection?.stop()
    }
}
